### PR TITLE
1.x: Make email field unique only for Users and Accounts

### PR DESCRIPTION
**Pull request by othercorey**
Closes https://github.com/cakephp/bake/issues/689

### CODE REVIEW

**Comment by othercorey** [comment]
Local tests pass.

File: src/Shell/Task/ModelTask.php [hunk 1]
```diff
@@ -842,9 +842,14 @@ public function getRules($model, array $associations)
             return [];
         }
 
+        $uniqueColumns = ['username', 'login'];
+        if (in_array($model->getAlias(), ['Users', 'Accounts'])) {
```
Comment (ADmad) [inline src/Shell/Task/ModelTask.php:846]: Wouldn't a better way would be to check for unique constraint for the field in the schema?

Comment (othercorey) [reply src/Shell/Task/ModelTask.php:846]: This is just backporting the snippet from Bake 2.x to satisfy the issue in Bake 1.x.

Comment (ADmad) [reply src/Shell/Task/ModelTask.php:846]: ok

Comment (othercorey) [reply src/Shell/Task/ModelTask.php:846]: The logic you're looking for exists below this.

Comment (othercorey) [reply src/Shell/Task/ModelTask.php:846]: If this isn't worth it for 1.x, then we can just close this.

**Comment by othercorey** [comment]
Closing due to no interest.

**Pull request closed without merging**